update_twiki.py

```diff
@@ -3,48 +3,104 @@
+                  extension text,                                                                                                                                                                                                             
+                  total_events integer NOT NULL,                                                                                                                                                                                              
+                  campaign text NOT NULL,                                                                                                                                                                                                     
+                  resp_group text NOT NULL,
+                  
+                  cross_section float NOT NULL,
+
+                  fraction_negative_weight float NOT NULL,
+                  
+                  target_num_events real NOT NULL   )''')
+
+    # Clear the table                                                                                                                                                                                                                         
     c.execute('DELETE FROM `twiki_samples`')
     conn.commit()
 
-    # Get all needed requests
+    # Get all needed requests                                                                                                                                                                                                                 
     twiki_samples_fall_18_candidates = mcm.get('requests', query='member_of_campaign=RunIIFall18*')
 
     total_events_threshold = 20000000
+    
+    file_twiki = open("MainSamplesFall18.txt", "w")
 
     for twiki_request in twiki_samples_fall_18_candidates:
 
         if(twiki_request['total_events']>total_events_threshold):
+            #Getting the cross_section value from xsdb
+            query = {'DAS': twiki_request['dataset_name']}
+            
+            search_rslt = xsdb_request.simple_search_to_dict(query)
+            cross_section = 1
+            frac_neg_wgts = 0
+            target_num_events = 0
+            if(len(search_rslt)>1):
+                search_rslt_ = search_rslt[1]
+                cross_section = float(search_rslt_[u'cross_section'])
+                frac_neg_wgts = float(search_rslt_[u'fraction_negative_weight'])
+            else:
+                try:
+                    cross_section = float(twiki_request['generator_parameters'][-1][u'cross_section'])
+                    frac_neg_wgts = float(twiki_request['generator_parameters'][-1][u'negative_weights_fraction'])
+                except:
+                    print(twiki_request['generator_parameters'])
+            def frac_not_minus(x):
```

Comment:
This function itself is not bad, but:
 1. Avoid defining functions in a loop
 2. This can be changed by just `max(0, x)`